File: catkin_ws/src/blinky/scripts/BlinkyAPI.py
#!/usr/bin/env python
import roslib; roslib.load_manifest('blinky')

import rospy

from blinky.msg import *
from blinky.srv import *

ledCount = 30

PLANNER_ID = 1
BATTERY_ID = 2

# Handles the ROS-specific communication with the service
# colors:       array of RGB
# blinkyID:     blinky segment the colors should be sent to
def send_colorList(colors, blinkyID):
    try:
        rospy.wait_for_service('Blinky')
        Blinky = rospy.ServiceProxy('Blinky', BlinkyService)
        res = Blinky(colors, blinkyID)

        if res.success != 0:
            print "Blinky request unsuccessful: %s"%res

    except Exception as e:
        print "Exception: %s"%e

def Planner_sendColors(colors):
    send_colorList(colors,1)

def Battery_sendColors(colors):
    send_colorList(colors,2)

File: catkin_ws/src/blinky/src/BlinkyAPI.cpp
#include "ros/ros.h"
#include <cstdlib>
#include <vector>

#include <blinky/BlinkyService.h>
#include <blinky/RGB.h>
#include "BlinkyAPI.h"

const int PLANNER_ID = 1;
const int BATTERY_ID = 2;

BlinkyClient::BlinkyClient(ros::NodeHandle node_handle)
{
        n = node_handle;
}

int8_t BlinkyClient::send_colorList(const std::vector<blinky::RGB>& colors, int8_t blinkyID)
{
	ros::ServiceClient client = n.serviceClient<blinky::BlinkyService>("Blinky");

	blinky::BlinkyService srv;
	srv.request.btColors = colors;
	srv.request.blinkyID = blinkyID;

	if (client.call(srv)) {
		int8_t res = srv.response.success;

		if (res) {
			ROS_INFO("Blinky request unsuccessful: %d", res);
		}

		return res;
	} else {
		ROS_ERROR("Failed to call service Blinky");
		return -1;
	}

	return 0;
}

int8_t BlinkyClient::Planner_sendColors(const std::vector<blinky::RGB>& colors)
{
	return send_colorList(colors, PLANNER_ID);
}

int8_t BlinkyClient::Battery_sendColors(const std::vector<blinky::RGB>& colors)
{
	return send_colorList(colors, BATTERY_ID);
}
